src/main/scala/org/allenai/pipeline/Producer.scala

```diff
@@ -125,7 +156,7 @@ trait CachingDisabled extends CachingEnabled {
   override def cachingEnabled: Boolean = false
 }
 
-class PersistedProducer[T, -A <: Artifact](
+class PersistedProducer[T, -A <: Artifact] private[pipeline] (
```

Comment:
It does.  Why do you subclass `PersistedProducer` instead of making a new class that behaves like `PersistedProducer`?